backends/advanced/src/advanced_omi_backend/routers/modules/obsidian_routes.py

```
@@ -0,0 +1,225 @@
+    # Check if already in RQ
+    try:
+        job = Job.fetch(job_id, connection=redis_conn)
+        status = job.get_status()
+        if status in ("queued", "started", "deferred", "scheduled"):
+             raise HTTPException(status_code=400, detail=f"Job already {status}")
+        
+        # If finished/failed, we could potentially restart? But for now let's say it's done.
+        raise HTTPException(status_code=400, detail=f"Job is in state: {status}")
+        
+    except NoSuchJobError:
+        raise HTTPException(status_code=404, detail="Job not found")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)